props/saber_caiwyn_buttons.h

```diff
@@ -0,0 +1,604 @@
+      }
+      strcpy(current_track_, next_track);
+#ifdef SAVE_TRACK
+      current_preset_.track = mkstr(current_track_);
+      current_preset_.Save();
+#endif
+      PlayTrack();
+    } else {
+      StartOrStopTrack();
+      StartOrStopTrack();
+    }
+  }
+
+#ifndef DISABLE_COLOR_CHANGE
+// Revert color change without saving (reset to Variation == 0)
+  void ResetColorChangeMode() {
+    if (!current_style()) return;
+    STDOUT << "Reset Color Variation" << "\n";
+    SetVariation(0);
+    STDOUT << "Color change mode done, variation = 0" << "\n";
+    SaberBase::SetColorChangeMode(SaberBase::COLOR_CHANGE_MODE_NONE);
+  }
+#endif
+
+  bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, modifiers)) {
+
+// Activate Saber
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_OFF):
+        On();
+        return true;
+
+// Activate Saber and Play Track
+#if NUM_BUTTONS > 2
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_OFF | BUTTON_AUX2):
+#else
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_OFF | BUTTON_AUX):
+#endif
+        On();
+        if (!track_player_) {
+          StartTrackPlayer();
+        }
+        return true;
+
+// Deactivate Saber
+#if NUM_BUTTONS > 2
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_ON | BUTTON_AUX2):
+#else
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_ON | BUTTON_AUX):
+#endif
+        if (!mode_volume_ && (SaberBase::GetColorChangeMode() == SaberBase::COLOR_CHANGE_MODE_NONE)) {
+          Off();
+          return true;
+        }
+        break;
+
+// Start or Stop Track
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_OFF):
+        if (track_player_) {
+          StopTrackPlayer();
+        } else {
+          StartTrackPlayer();
+        }
+        return true;
+
+// Next Preset/Track
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_MEDIUM, MODE_OFF):
+#ifndef DISABLE_BEEPS
+        beeper.Beep(0.1, 2000);
+#endif
+        if (track_player_) {
+          NextTrack();
+        } else {
+          next_preset();
+          strcpy(current_track_,current_preset_.track.get());
+          track_mode_ = PLAYBACK_ONCE;
+        }
+        return true;
+
+// Battery Level
+      case EVENTID(BUTTON_AUX, EVENT_SECOND_HELD_MEDIUM, MODE_OFF):
+        if (!track_player_on_ && !wav_player->isPlaying()) {
+#ifdef CUSTOM_BATTERY_LEVEL
+          sound_library_.SayBatteryLevel();
+          sound_library_.SayNumber(battery_monitor.battery_percent(), SAY_WHOLE);
+          sound_library_.SayPercent();
+#else
+          talkie.Say(spBATTERYLEVEL);
+          talkie.SayNumber((int)floorf(battery_monitor.battery_percent()));
+          talkie.Say(spPERCENT);
+#endif
+          SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
+          return true;
+        }
+        break;
+
+// Track Playback Mode
+      case EVENTID(BUTTON_AUX, EVENT_SECOND_CLICK_SHORT, MODE_OFF):
+        switch (track_mode_) {
```

Comment:
Probably better to break this out into a helper function.

Reply:
I can do that.

Reply:
Done.